clients/src/main/java/org/apache/kafka/common/utils/ByteUtils.java

```diff
@@ -413,15 +424,20 @@ public static int sizeOfVarint(int value) {
      * Number of bytes needed to encode a long in variable-length format.
      *
      * @param value The signed value
+     * @see #sizeOfUnsignedVarint(int)
      */
     public static int sizeOfVarlong(long value) {
         long v = (value << 1) ^ (value >> 63);
-        int bytes = 1;
-        while ((v & 0xffffffffffffff80L) != 0L) {
-            bytes += 1;
-            v >>>= 7;
-        }
-        return bytes;
+
+        // For implementation notes @see #sizeOfUnsignedVarint(int)
+
+        // Similar logic is applied to allow for 64bit input -> 1-9byte output.
+
+        // return (70 - leadingZeros) / 7 + leadingZeros / 64;
```

Comment:
Nit: not sure we need the empty lines between comments here, seems pretty readable without it.